File: src/main/java/ar/edu/unq/epers/unidad5/service/PersonajeService.java
package ar.edu.unq.epers.unidad5.service;

import ar.edu.unq.epers.unidad5.model.Item;
import ar.edu.unq.epers.unidad5.model.Personaje;
import ar.edu.unq.epers.unidad5.model.PesoPromedio;

import java.util.List;

public interface PersonajeService {
    Personaje crear(Personaje personaje);
    void crearTodos(List<Personaje> personajes);
    void recoger(String personajeId, Item item);
    List<Personaje> recuperarPorNombre(String nombre);
    List<Personaje> recuperarPorVida(int vida);
    List<Personaje> recuperarPorRangoDeVida(int min, int max);
    List<Item> recuperarItemsDePersonaje(String nombre);
    List<PesoPromedio> obtenerPesoPromedio(List<String> nombres);
    void eliminarTodo();
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/service/PersonajeService.kt
package ar.edu.unq.epers.unidad5.service

import ar.edu.unq.epers.unidad5.model.Item
import ar.edu.unq.epers.unidad5.model.Personaje
import ar.edu.unq.epers.unidad5.model.PesoPromedio

interface PersonajeService {
    fun crear(personaje: Personaje): Personaje
    fun crearTodos(personajes: List<Personaje>)
    fun recoger(personajeId: String, item: Item)
    fun recuperarPorNombre(nombre: String): List<Personaje>
    fun recuperarPorVida(vida: Int): List<Personaje>
    fun recuperarPorRangoDeVida(min: Int, max: Int): List<Personaje>
    fun recuperarItemsDePersonaje(nombre: String): List<Item>
    fun obtenerPesoPromedio(nombres: List<String>): List<PesoPromedio>
    fun eliminarTodo()
}